packages/runtime-core/src/apiWatch.ts

```diff
@@ -451,11 +451,7 @@ export function traverse(value: unknown, seen?: Set<unknown>) {
   seen.add(value)
   if (isRef(value)) {
     traverse(value.value, seen)
-  } else if (isArray(value)) {
-    for (let i = 0; i < value.length; i++) {
-      traverse(value[i], seen)
-    }
-  } else if (isSet(value) || isMap(value)) {
+  } else if (isSet(value) || isMap(value) || isArray(value)) {
```

Comment:
Perhaps this is for the pursuit of extreme performance considerations? I am not sure 🧐
![image](https://github.com/vuejs/core/assets/32354856/01b2564e-7a2b-41f7-af34-b9ef94732cec)
https://levelup.gitconnected.com/which-is-faster-for-for-of-foreach-loops-in-javascript-18dbd9ffbca9

Reply:
I thought about this, but only when the amount of data is huge, for loop is faster than forEach.